test/Control/Carrier/DiagnosticsSpec.hs

```diff
@@ -6,7 +6,8 @@ import Control.Carrier.Telemetry (withoutTelemetry)
 import Control.Effect.Diagnostics (ToDiagnostic (..), fatal)
 import Data.Text (Text)
 import Effect.Exec (ExitCode (..))
-import Effect.Logger (ignoreLogger, logInfo, pretty)
+import Effect.Logger (ignoreLogger, logInfo)
```

Comment:
This is a scarily small ratio of test change to code change. 

It could be that there are a bunch of tests for error cases but the changes in this PR don't affect those tests. In that case, the error code is being exercised by the tests and all is good. 

If there are no tests on most of these errors currently, we should fix this up. I'm fine with doing that in a separate PR, as this one is huge.

If you've thought about this and come to the conclusion that this is too hard, that's reasonable, but it should be documented in the PR description.

Reply:
Yea I'm going to create a separate ticket to add more tests. I agree, I was pretty shocked when I saw the state of our test coverage here.